### PR TITLE
Add code coverage for ToolStripMenuItem

### PR DESCRIPTION
**Pull request by Zheng-Li01**
related https://github.com/dotnet/winforms/issues/10453

# Proposed changes
Add unit tests for ToolStripMenuItem to test its properties & methods & events
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/11887)

### CODE REVIEW

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripMenuItemTests.cs [hunk 1]
```diff
@@ -183,6 +183,225 @@ public static IEnumerable<object[]> CultureInfo_Shortcut_TestData()
         yield return new object[] { new CultureInfo("de-DE"), new CultureInfo("de-DE"), "Strg+Umschalttaste+K" };
     }
 
+    [WinFormsFact]
+    public void ToolStripMenuItem_Ctors_ShouldInitializeCorrectly()
+    {
+        using Image image = new Bitmap(10, 10);
```
Comment (Epica3055) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripMenuItemTests.cs:189]:
```suggestion
        using Image image = new(10, 10);
```

Comment (Epica3055) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripMenuItemTests.cs:189]: and other places

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripMenuItemTests.cs [hunk 2]
```diff
@@ -183,6 +183,225 @@ public static IEnumerable<object[]> CultureInfo_Shortcut_TestData()
...
+        const string text = "Test Item";
+        const string name = "TestName";
+        EventHandler onClick = (sender, e) => { };
+
+        using ToolStripMenuItem item = new(text, image, onClick, name);
+
+        item.Text.Should().Be(text);
+        item.Image.Should().Be(image);
+        item.Name.Should().Be(name);
+        item.Click += onClick;
```
Comment (LeafShi1) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripMenuItemTests.cs:214]: This event has been registered in the constructor, so what we should verify here is whether this event can be triggered correctly

```
        bool wasClicked = false;
        EventHandler onClick = (sender, e) => wasClicked = true;
        using ToolStripMenuItem item = new(text, image, onClick, name);
       // item.Click += onClick;

        item.TestAccessor().Dynamic.OnClick(null);
        wasClicked.Should().BeTrue();

```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripMenuItemTests.cs [hunk 3]
```diff
@@ -183,6 +183,225 @@ public static IEnumerable<object[]> CultureInfo_Shortcut_TestData()
...
+        using Image image = new Bitmap(10, 10);
+        const string text = "Test Item";
+        EventHandler onClick = (sender, e) => { };
+        Keys shortcutKeys = Keys.Control | Keys.A;
+
+        using ToolStripMenuItem item = new(text, image, onClick, shortcutKeys);
+
+        item.Text.Should().Be(text);
+        item.Image.Should().Be(image);
+        item.Click += onClick;
```
Comment (LeafShi1) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripMenuItemTests.cs:229]: Same issue as above

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripMenuItemTests.cs (hunk 1 above)
Comment (LeafShi1) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripMenuItemTests.cs:189]: It should be `using Bitmap image = new(10, 10);`

**Review by LeafShi1 — COMMENTED** [review]
Can you please add `#nullable enable` in this file

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripMenuItemTests.cs [hunk 4]
```diff
@@ -183,6 +185,227 @@ public static IEnumerable<object[]> CultureInfo_Shortcut_TestData()
...
+        accessor.DeviceDpi = 120;
+
+        ((object)accessor.t_indeterminateCheckedImage).Should().BeNull();
+        ((object)accessor.t_checkedImage).Should().BeNull();
+    }
+
+    [WinFormsFact]
+    public void ToolStripMenuItem_CheckedChanged_InvokeHandler()
+    {
+        ToolStripMenuItem item = new();
```
Comment (Epica3055) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripMenuItemTests.cs:376]: why not `using` here

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripMenuItemTests.cs [hunk 5]
```diff
@@ -183,6 +185,227 @@ public static IEnumerable<object[]> CultureInfo_Shortcut_TestData()
...
+        item.CheckedChanged += handler;
+
+        item.Checked = true;
+        eventCalled.Should().BeTrue();
+
+        eventCalled = false;
+        item.Checked = false;
+        eventCalled.Should().BeTrue();
+
+        item.CheckedChanged -= handler;
```
Comment (Epica3055) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripMenuItemTests.cs:389]: I don't think this is necessary.

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripMenuItemTests.cs [hunk 6]
```diff
@@ -183,6 +185,227 @@ public static IEnumerable<object[]> CultureInfo_Shortcut_TestData()
...
+    }
+
+    [WinFormsFact]
+    public void ToolStripMenuItem_CheckStateChanged_InvokeHandler()
+    {
+        ToolStripMenuItem item = new();
```
Comment (Epica3055) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripMenuItemTests.cs:395]: why not `using` here

**Review by Epica3055 — DISMISSED** [review]

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripMenuItemTests.cs [hunk 7]
```diff
@@ -183,6 +185,225 @@ public static IEnumerable<object[]> CultureInfo_Shortcut_TestData()
         yield return new object[] { new CultureInfo("de-DE"), new CultureInfo("de-DE"), "Strg+Umschalttaste+K" };
     }
 
+    [WinFormsFact]
+    public void ToolStripMenuItem_Ctor_ShouldInitializeCorrectly()
+    {
+        using Bitmap image = new(10, 10);
+        const string text = "Test Item";
```
Comment (lonitra) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripMenuItemTests.cs:192]: Normally const are defined at the beginning of a class. We don't need to specify one in the method.
```suggestion
        string text = "Test Item";
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripMenuItemTests.cs [hunk 8]
```diff
@@ -183,6 +185,225 @@ public static IEnumerable<object[]> CultureInfo_Shortcut_TestData()
...
+        itemWithTextAndImage.Text.Should().Be(text);
+        itemWithTextAndImage.Image.Should().Be(image);
+    }
+
+    [WinFormsFact]
+    public void ToolStripMenuItem_Ctor_WithTextImageOnClickName_ShouldInitializeCorrectly()
+    {
+        using Bitmap image = new(10, 10);
+        const string text = "Test Item";
+        const string name = "TestName";
```
Comment (lonitra) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripMenuItemTests.cs:208]:
```suggestion
        string text = "Test Item";
        string name = "TestName";
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripMenuItemTests.cs [hunk 9]
```diff
@@ -183,6 +185,225 @@ public static IEnumerable<object[]> CultureInfo_Shortcut_TestData()
...
+        item.Name.Should().Be(name);
+        item.TestAccessor().Dynamic.OnClick(null);
+        wasClicked.Should().BeTrue();
+    }
+
+    [WinFormsFact]
+    public void ToolStripMenuItem_Ctor_WithTextImageOnClickShortcutKeys_ShouldInitializeCorrectly()
+    {
+        using Bitmap image = new(10, 10);
+        const string text = "Test Item";
```
Comment (lonitra) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripMenuItemTests.cs:225]:
```suggestion
        string text = "Test Item";
```

**Review by lonitra — COMMENTED** [review]
Small comment about const otherwise LGTM

**Pull request merged**